### ISSUE TITLE
Allow audio autoplay only with user interaction

### ISSUE THREAD

**Issue by csagan5** [bromite/bromite#804]
As a spin-off of #804 we are investigating the "document activation" behaviour of the autoplay functionality for audio.

> **UnifiedAutoplayConfig**
> Add a toggle to the sound content settings page to toggle unified autoplay. Also adds a pref to store the user's preference.

This is the code handling the policy:
```
 switch (GetAutoplayPolicy()) {
    case AutoplayPolicy::Type::kNoUserGestureRequired:
      NOTREACHED();
      break;
    case AutoplayPolicy::Type::kUserGestureRequired:
      DCHECK(window->GetFrame());
      DCHECK(window->GetFrame()->IsCrossOriginToMainFrame());
      window->AddConsoleMessage(MakeGarbageCollected<ConsoleMessage>(
          mojom::ConsoleMessageSource::kOther,
          mojom::ConsoleMessageLevel::kWarning,
          "The AudioContext was not allowed to start. It must be resumed (or "
          "created) from a user gesture event handler. https://goo.gl/7K7WLu"));
      break;
    case AutoplayPolicy::Type::kDocumentUserActivationRequired:
      window->AddConsoleMessage(MakeGarbageCollected<ConsoleMessage>(
          mojom::ConsoleMessageSource::kOther,
          mojom::ConsoleMessageLevel::kWarning,
          "The AudioContext was not allowed to start. It must be resumed (or "
          "created) after a user gesture on the page. https://goo.gl/7K7WLu"));
      break;
  }
```

By default `GetAutoplayPolicy()` returns `AutoplayPolicy::Type::kDocumentUserActivationRequired`; we will switch it to `kUserGestureRequired`.

**Comment by uazo** [comment]
here related code

https://source.chromium.org/chromium/chromium/src/+/master:content/browser/web_contents/web_contents_impl.cc;l=2427
https://source.chromium.org/chromium/chromium/src/+/master:media/base/media_switches.cc;l=694

Since `kUnifiedAutoplay` is `FEATURE_ENABLED_BY_DEFAULT`
https://source.chromium.org/chromium/chromium/src/+/master:media/base/media_switches.cc;l=433

autoplay default seems be `autoplay::kDocumentUserActivationRequiredPolicy` (for sounds and videos)

**Comment by csagan5** [comment]
Ok, then I will try to use `kUserGestureRequired` as a default and see how it works.

**Comment by csagan5** [comment]
I will switch `kUnifiedAutoplay` to disabled for next release.

**Comment by csagan5** [comment]
Fixed in `87.0.4280.131`.
